cdap-explore/src/main/java/co/cask/cdap/explore/executor/ExploreMetadataHttpHandler.java

```diff
@@ -67,7 +67,8 @@ public ExploreMetadataHttpHandler(ExploreService exploreService) {
 
   @GET
   @Path("data/explore/tables")
-  public void getTables(HttpRequest request, HttpResponder responder, @QueryParam("db") String databaseName) {
+  public void getTables(HttpRequest request, HttpResponder responder,
+                        @QueryParam("db") String databaseName, @PathParam("namespace-id") String namespaceId) {
```

Comment:
Ok i've figured out what this is.

The UI uses this endpoint to list all explorable datasets (I think its an undocumented endpoint). If `db` is `null` then `ExploreService` that handles this call will automatically look through all `db`s. So the UI never passes in the `db` value and it always shows all explorable datasets. Correct thing to do would be to remove the `QueryParam` and instead of `db` pass in `namespaceId` (since thats what we'll use as the `db` from now onwards).